src/evaluate/evaluator/base.py

```diff
@@ -350,7 +350,7 @@ def prepare_pipeline(
                 pipe = model_or_pipeline
             if tokenizer is not None and feature_extractor is not None:
                 logger.warning("Ignoring the value of the preprocessor argument (`tokenizer` or `feature_extractor`).")
-        if pipe.task != self.task:
+        if (pipe.task != self.task) and not (self.task == "translation" and pipe.task.startswith("translation")):
```

Comment:
Why is this necessary?

Reply:
Because for translation the task in the pipeline is `translation_xx_to_yy` where the placeholders are the languages.